File: server/src/Domain/Entities/Submission.cs
using Domain.Enums;

namespace Domain.Entities;

public class Submission : BaseAuditableEntity
{
    public string AnswerText { get; set; } = string.Empty;
    public DateTime SubmittedAt { get; set; }
    public SubmissionStatus Status { get; set; } = SubmissionStatus.Submitted;

    public int? Marks { get; set; }
    public string? Feedback { get; set; }
    public DateTime? GradedAt { get; set; }
    public Guid? GradedById { get; set; }
    public User? GradedBy { get; set; }

    public Guid AssignmentId { get; set; }
    public Assignment Assignment { get; set; } = null!;

    public Guid StudentId { get; set; }
    public User Student { get; set; } = null!;
}

File: client/src/types/submission.ts
export type SubmissionStatus = "Submitted" | "Graded" | "Returned";

export interface SubmissionDto {
  id: string;
  assignmentId: string;
  assignmentTitle: string;
  maxMarks: number;
  deadline: string;
  studentId: string;
  studentName: string;
  answerText: string;
  submittedAt: string;
  status: SubmissionStatus;
  marks: number | null;
  feedback: string | null;
  gradedAt: string | null;
  gradedByName: string | null;
}
